front/app/utils/moduleUtils.ts

```diff
@@ -528,28 +529,46 @@ export const insertConfiguration = <T extends { name: string }>({
   configuration,
   insertAfterName,
   insertBeforeName,
+  // if the inserted element contains data that's updated,
+  // we can force the element to be reinserted
+  reinsertAfterUpdate,
 }: InsertConfigurationOptions<T>) => (items: T[]): T[] => {
   const itemAlreadyInserted = items.some(
     (item) => item.name === configuration.name
   );
-  const foundIndex = items.findIndex(
+  // index of item where we need to insert before/after
+  const referenceIndex = items.findIndex(
     (item) => item.name === (insertAfterName || insertBeforeName)
   );
   const insertIndex = clamp(
-    insertAfterName ? foundIndex + 1 : foundIndex - 1,
+    // if number is outside of lower and upper, it picks
```

Comment:
Maybe I am missing something here and there is a good reason but why can't we achieve this with regular state updates?

Reply:
What specifically would you try to achieve with regular state?

Reply:
What I meant was that instead of inserting and reinserting the tabs we can have a mechanism for the already inserted tabs to re-render on data change. More details in my other comment above!